src/Routes/helpers.ts

```diff
@@ -28,6 +29,8 @@ export type Route = {
   hide?: boolean
   breadcrumb?: string | React.ReactNode
   category?: RouteCategory
+  mobileNav?: boolean
+  priority?: number
```

Comment:
Definitely a nitpick but if you make this a discriminated union i.e
```suggestion
& ({
  mobileNav: boolean
  priority: number
}) | {
  mobileNav?: never
  priority?: never
}
```

You won't have to opt chain